pkg/component/certmanagement/deployment.go

```diff
@@ -0,0 +1,400 @@
+				},
+				{
+					APIGroups: []string{"dns.gardener.cloud"},
+					Resources: []string{"dnsentries"},
+					Verbs:     []string{"get", "list", "update", "watch", "create", "delete"},
+				},
+			},
+		}
+		roleBinding = &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      role.Name,
+				Namespace: d.namespace,
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      rbacv1.ServiceAccountKind,
+					Name:      serviceAccount.Name,
+					Namespace: serviceAccount.Namespace,
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "Role",
+				Name:     role.Name,
+			},
+		}
+
+		deployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName,
+				Namespace: d.namespace,
+				Labels: gardenerutils.MergeStringMaps(getDeploymentLabels(), map[string]string{
+					resourcesv1alpha1.HighAvailabilityConfigType: resourcesv1alpha1.HighAvailabilityConfigTypeController,
+				}),
+			},
+			Spec: appsv1.DeploymentSpec{
+				Replicas:             ptr.To[int32](1),
+				RevisionHistoryLimit: ptr.To[int32](5),
+				Selector:             &metav1.LabelSelector{MatchLabels: getDeploymentLabels()},
+				Strategy:             appsv1.DeploymentStrategy{Type: appsv1.RecreateDeploymentStrategyType},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: gardenerutils.MergeStringMaps(getDeploymentLabels(), map[string]string{
+							v1beta1constants.LabelNetworkPolicyToDNS:              v1beta1constants.LabelNetworkPolicyAllowed,
+							v1beta1constants.LabelNetworkPolicyToRuntimeAPIServer: v1beta1constants.LabelNetworkPolicyAllowed,
+							v1beta1constants.LabelNetworkPolicyToPublicNetworks:   v1beta1constants.LabelNetworkPolicyAllowed,
+							v1beta1constants.LabelNetworkPolicyToPrivateNetworks:  v1beta1constants.LabelNetworkPolicyAllowed,
+						}),
+					},
+					Spec: corev1.PodSpec{
+						ServiceAccountName: serviceAccount.Name,
+						Containers: []corev1.Container{{
+							Name:            containerName,
+							Image:           d.image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Args: []string{
+								"--name=cert-controller-manager",
+								fmt.Sprintf("--dns-namespace=%s", d.namespace),
+								"--use-dnsrecords",
+								fmt.Sprintf("--issuer.issuer-namespace=%s", d.namespace),
+								fmt.Sprintf("--server-port-http=%d", serverPort),
+								fmt.Sprintf("--default-rsa-private-key-size=%d", rsaPrivateKeySize),
+							},
+							LivenessProbe: &corev1.Probe{
+								ProbeHandler: corev1.ProbeHandler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path:   "/healthz",
+										Port:   intstr.FromInt32(serverPort),
+										Scheme: "HTTP",
+									},
+								},
+								InitialDelaySeconds: 30,
+								TimeoutSeconds:      5,
+								PeriodSeconds:       10,
+								SuccessThreshold:    1,
+								FailureThreshold:    3,
+							},
+							Resources: corev1.ResourceRequirements{
```

Comment:
We could create a VPA resource for this deployment too.

Reply:
The cert-management component has minimal work load, no need for scaling.